src/main/groovy/edu/oregonstate/mist/metaxe/health/MetaXEHealthCheck.groovy

```diff
@@ -21,7 +23,18 @@ class MetaXEHealthCheck extends HealthCheck {
             return Result.unhealthy("xeAppsFilePath is null")
         }
         if (xeAppsFile.exists()) {
-            Result.healthy()
+            Map<String, Attributes> allAttributes
+            try {
+                allAttributes = XEAppDAO.getAllAttributes(xeAppsFile)
+            } catch (Exception e) {
+                return Result.unhealthy(e)
+            }
+
+            if (allAttributes.isEmpty()) {
+                Result.unhealthy("No data found in xeAppsFile")
+            } else {
+                Result.healthy()
```

Comment:
I think you could merge this logic into `try-catch` block by throwing an exception.